packages/strapi-plugin-i18n/admin/src/reducers.js

```diff
@@ -0,0 +1,53 @@
+import pluginId from './pluginId';
+import {
+  LOAD_LOCALES,
+  RESOLVE_LOCALES,
+  ADD_LOCALE,
+  DELETE_LOCALE,
+  UPDATE_LOCALE,
+} from './constants';
+
+export const initialState = {
+  isLoading: false,
+  locales: [],
+};
+
+const localeReducer = (state = initialState, action) => {
```

Comment:
Can you move this reducer in the hooks/useLocales  folder so we know which component is mainly using it?

Reply:
Also, can you make the state immutable using [`immer`](https://immerjs.github.io/immer/docs/introduction)?

Reply:
> Can you move this reducer in the hooks/useLocales folder so we know which component is mainly using it?

I was thinking about that, but since the reducer is not only about useLocale for is transversal to i18n in general, I didn't know where to put it 😬

Reply:
By default I would put it in the hook folder.